src/Integration.Vsix/InfoBar/InfoBarManager.cs

```diff
@@ -214,39 +248,50 @@ private class PrivateInfoBarWrapper : IInfoBar, IVsInfoBarUIEvents
             private uint? cookie;
             private bool isClosed;
 
-            public PrivateInfoBarWrapper(IVsWindowFrame frame, IVsInfoBarUIElement uiElement)
+            public PrivateInfoBarWrapper(IVsInfoBarHost host, IVsInfoBarUIElement uiElement)
             {
-                Debug.Assert(frame != null);
+                Debug.Assert(host != null);
                 Debug.Assert(uiElement != null);
 
+                this.Host = host;
                 this.InfoBarUIElement = uiElement;
-                this.Frame = frame;
-
-                this.Frame.ShowNoActivate();
 
                 this.Advise();
             }
 
-            internal IVsInfoBarUIElement InfoBarUIElement { get; }
+            private IVsInfoBarUIElement InfoBarUIElement { get; }
 
-            internal IVsWindowFrame Frame { get; }
+            private IVsInfoBarHost Host { get; }
 
             #region IInfoBarEvents
             public event EventHandler<InfoBarButtonClickedEventArgs> ButtonClick;
             public event EventHandler Closed;
 
-            public void Close()
+            private void CloseUIElement()
             {
                 if (this.isClosed)
                 {
                     return;
                 }
 
                 ThreadHelper.ThrowIfNotOnUIThread();
+                // This will fire the Closed event
                 this.InfoBarUIElement.Close();
 
                 Debug.Assert(!this.cookie.HasValue, "Expected to be unadvised");
             }
+
+            public void Close()
+            {
+                CloseUIElement();
+
+                ThreadHelper.ThrowIfNotOnUIThread();
+                if (Host != null )
+                {
+                    Host.RemoveInfoBar(InfoBarUIElement);
+                }
+            }
```

Comment:
This is the method that is called by `InfoBarManager.Detach` in lines 102-109. It is functionally the same.